src/VecSim/spaces/CMakeLists.txt

```diff
@@ -10,14 +10,17 @@ include(CheckCXXCompilerFlag)
 project(VectorSimilarity_Spaces)
 
 # TODO: Remove this once cpu_features get support for M1
-if(NOT APPLE)
-	include(${root}/cmake/cpu_features.cmake)
-elseif(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64")
+if((NOT APPLE) OR (CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64"))
```

Comment:
No, because an Apple machine with Intel CPU supports the optimizations as well. only `ARM` machines cannot.

Reply:
so isn't checking x86_64 enough?

Reply:
here, by checking `NOT APPLE` we support Linux on ARM (like the Ubuntu Linux we have on CI), which has support for the runtime CPU features. Currently, it doesn't help much since we don't have optimizations for ARM but in general, we want to include the runtime CPU feature to all the OSs and architectures we have basic support for.